pkg/controller/migplan/storage.go

```diff
@@ -124,6 +132,36 @@ func (r ReconcileMigPlan) ensureVSL(client k8sclient.Client, storage *migapi.Mig
 	return true, nil
 }
 
+// Create the velero BSL cloud secret has been created.
+// Returns `true` when ensured.
+func (r ReconcileMigPlan) ensureCloudSecret(client k8sclient.Client, storage *migapi.MigStorage) (bool, error) {
+	newSecret, err := storage.BuildCloudSecret(r)
```

Comment:
I'm a little unclear on the workflow here. From above, `BuildCloudSecret` will return an error `CredSecretNotFound` if it doesn't exist yet (as a result of the embedded `UpdateCloudSecret` call). If the secret already exists, all of this will return successfully, but on first run, it looks like this will error out rather than creating the secret.

Reply:
The `UpdateCloudSecret()` will return an error when the _source_ `CredsSecretRef` secret does not exist or invalid.  In practice this error will rarely be raised because `ensureCloudSecret()` is only called by `Reconcile()` after validation has passed.  This error handling is for robustness and to handle race conditions whereby the `CredsSecretRef` secret is deleted after validation is performed and the `ensureCloudSecret()` is called.

Reply:
As we discussed earlier today, it's much clearer now with the renaming.